packages/discordx/src/decorators/classes/DSimpleCommand.ts

```diff
@@ -203,79 +204,83 @@ export class DSimpleCommand extends Method {
             .filter((op) => op?.length)
             .map((op) => op.trim());
 
-    return this.options
-      .sort((a, b) => (a.index ?? 0) - (b.index ?? 0))
-      .map((op, index) => {
-        // only digits
-        const id = args[index]?.replace(/\D/g, "");
-
-        // undefined
-        if (!args[index]?.length) {
-          return undefined;
-        }
-
-        // Boolean
-        if (op.type === "BOOLEAN") {
-          return Boolean(args[index]);
-        }
-
-        // Number
-        if (op.type === "NUMBER" || op.type === "INTEGER") {
-          return Number(args[index]);
-        }
-
-        // Channel | undefined
-        if (op.type === "CHANNEL") {
-          if (!id?.length) {
-            return undefined;
-          }
-          return command.message.guild?.channels.resolve(id);
-        }
+    return Promise.all(
+      this.options
+        .sort((a, b) => (a.index ?? 0) - (b.index ?? 0))
+        .map((op, index) => {
+          // only digits
+          const id = args[index]?.replace(/\D/g, "");
 
-        // Role | undefined
-        if (op.type === "ROLE") {
-          if (!id?.length) {
+          // undefined
+          if (!args[index]?.length) {
             return undefined;
           }
 
-          return command.message.guild?.roles.resolve(id);
-        }
+          // Boolean
+          if (op.type === "BOOLEAN") {
+            return Boolean(args[index]);
+          }
 
-        // GuildMember | User | undefined
-        if (op.type === "USER") {
-          if (!id?.length) {
-            return undefined;
+          // Number
+          if (op.type === "NUMBER" || op.type === "INTEGER") {
+            return Number(args[index]);
           }
 
-          if (command.message.channel.type === "DM") {
-            return command.message.client.user?.id === id
-              ? command.message.client.users.resolve(id)
-              : command.message.author;
+          // Channel | undefined
+          if (op.type === "CHANNEL") {
+            if (!id?.length) {
+              return undefined;
+            }
+            return command.message.guild?.channels
+              .fetch(id)
+              .catch((err) => err);
```

Comment:
Catching errors and just returning them like normal data is not good practice. Let it throw the error back to the user. Same goes for other catches below.
If it's because you can't catch the error from the command's function, I'd say just catch it and return the original arg string.

@VictoriqueMoe what do you think? Swallowing the error is not great either..

Reply:
that is correct, but this is not a normal execution, in simple command a user can pass anything, for example instead of valid user id, a user pass ``abc``, this will crash the bot.

If we throw, means we are forcing devs to use try catch on ``client.executeCommand``, which imo not a good practice and passing it to handler is good, because then the developer can decide, what to do with it.

also before this update, it was ``USER | null`` because it was using djs catch, now it's the same but using ``fetch`` method

Reply:
I get that, but idk if this is the best way to handle it, since the user is still losing the input... Let's see what Vic has to say about it.

Reply:
already discussed with her in discord, she kinda agreed after knowing, why we shouldn't throw it.

Reply:
Yeah, I figured she wouldn't be too happy about it either lmao
Oh well, this is probably the best we'll get.

Reply:
> Yeah, I figured she wouldn't be too happy about it either lmao Oh well, this is probably the best we'll get.

I agree we should throw it, however, if returning an Error type as a union, it forces the user to do an instanceof check for Error. i don't like it, but it's the best we will get here :)